testplan/runnable/base.py

```diff
@@ -1165,31 +1170,91 @@ def _pre_exporters(self):
                 self._result.test_report
             )
 
-    def _invoke_exporters(self):
+    def _run_exporter(
+        self,
+        exporter: Exporter,
+        source: TestReport,
+        export_context: ExportContext,
+    ) -> ExporterResult:
+        """
+        Wraps an exporter run and handles exceptions.
+
+        :param exporter: exporter to run
+        :param source: Testplan report to export
+        :param export_context: information about other exporters
+        :return: ExporterResult object containing information about the actual exporter object and its possible output
+        """
+
+        traceback_string = None
+        exp_result = None
+        try:
+            exp_result = exporter.export(
+                source=source,
+                export_context=export_context,
+            )
+        except TypeError:
+            # TODO: Remove this except section in the future
+            self.logger.warning(
+                (
+                    "Exporter '%s' does not have keyword_argument 'export_context'! "
+                    "This will be prohibited in the future, please modify your custom exporter!"
+                ),
+                exporter,
+            )
+            exp_result = exporter.export(
+                source=source,
+            )
+        except Exception:
+            traceback_string = traceback.format_exc()
+        finally:
+            if isinstance(exp_result, ExporterResult):
+                exp_result.traceback = traceback_string
+            else:
+                if not traceback_string:
+                    # TODO: Move this part to the Exception section and remove the else block
+                    # We probably did not fail during the export
+                    self.logger.warning(
+                        (
+                            "Exporter '%s' returns with not an ExporterResult object! "
+                            "This will be prohibited in the future, please modify your custom exporter!"
+                        ),
+                        exporter,
+                    )
+                exp_result = ExporterResult(
+                    exporter=exporter,
+                    traceback=traceback_string,
+                    result={"unknown": exp_result},
```

Comment:
since we cannot make any assumptions here, i'd suggest we use `Any` as result type

actually, i think we can just have `export` methods returning `Any` since we need to wrap the exception and actual result here nevertheless. am i missing anything here?

Reply:
We would like to move the users to use ExportContest and ExporterResult in their custom exporters. The code will handle if not and will show a warning message.